.githooks/pre-commit

```diff
@@ -0,0 +1,30 @@
+#!/bin/sh
+
+staged_go_files=$(git diff --cached --name-only | grep ".go$")
```

Comment:
Using make targets would be better but good for now

Reply:
What I mean is that we should have go vet make target etc.

Reply:
a quick win as I'm seeing alot of PRs failing for formatting errors. I'm not familiar with make targets. will have a look into them at some stage